encoding/basex/stream_test.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2017 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package basex
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type fakeReader struct {
+	b   byte
+	n   int
+	err error
+}
```

Comment:
Kind of similar to `FakeIOErrorReader` in `go/chat/signencrypt/codec_test.go`. (Not similar enough to deduplicate though.)